agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/HttpServerTracer.java

```diff
@@ -218,23 +234,24 @@ private void setStatus(Span span, int status) {
     span.setStatus(HttpStatusConverter.statusFromHttpStatus(status));
   }
 
-  protected abstract Integer peerPort(REQUEST request);
+  protected abstract Integer peerPort(CONNECTION connection);
 
-  protected abstract String peerHostIP(REQUEST request);
+  protected abstract String peerHostIP(CONNECTION connection);
 
   protected abstract HttpTextFormat.Getter<REQUEST> getGetter();
 
   protected abstract URI url(REQUEST request) throws URISyntaxException;
 
   protected abstract String method(REQUEST request);
 
-  /** Stores given context in the given request in implementation specific way. */
-  protected abstract void attachContextToRequest(Context context, REQUEST request);
+  /** Stores given context in the given storage in implementation specific way. */
```

Comment:
trying to differentiate more between the different storages
```suggestion
  /** Stores given context in the request-response-loop storage in implementation specific way. */```